packages/xod-client-electron/src/app/utils.js

```diff
@@ -12,3 +15,42 @@ export const errorToPlainObject = R.when(
     R.identity,
   ])
 );
+
+// :: String -> Maybe Path
+const maybePath = R.ifElse(
+  path.isAbsolute.bind(path),
+  Maybe.of,
+  Maybe.Nothing
+);
+
+/**
+ * It provides one iterface for getting file path, that
+ * should be opened on start-up (if User opens associated file),
+ * on any platform:
+ * - Windows & Linux: get it from process.argv
+ * - MacOS: get it from fired events
+ *
+ * It accepts an app and returns a getter function,
+ * that will return `Promise Path Null`.
+ */
+// :: App -> (_ -> Promise Path Null)
+export const getFilePathToOpen = (app) => {
+  // Windows & Linux
+  let pathToOpen = R.compose(
+    R.chain(maybePath),
+    Maybe
+  )(process.argv[1]);
```

Comment:
Isn’t `process.argv[1]` fragile?

- `electron . /path/to/project`
- `node /path/to/electron /path/to/xod /path/to/project`
- `xod-client-electron /path/to/project`

Does `yarn start:electron -- -- /path/to/project` work on your machine?